cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -119,32 +119,53 @@ func (o *ScaleUpOrchestrator) ScaleUp(
 		}
 	}
 
+	var nodeGroupPoolList [][]cloudprovider.NodeGroup
+	// Add cloudprovider nodegroup pool into the list.
+	nodeGroupPoolList = append(nodeGroupPoolList, nodeGroups)
+
+	if o.processors != nil && o.processors.NodeGroupPoolListProcessor != nil {
+		var err error
+		nodeGroupPoolList, nodeInfos, err = o.processors.NodeGroupPoolListProcessor.Process(o.autoscalingContext, nodeGroupPoolList, nodeInfos, unschedulablePods)
+		if err != nil {
+			return scaleUpError(&status.ScaleUpStatus{}, errors.ToAutoscalerError(errors.InternalError, err))
+		}
+	}
+
```

Comment:
I feel like we should be using only either `NodeGroupListProcessor` or `NodeGroupPoolListProcessor` in a simple loop. It seems less ambiguous from the implementation perspective. Something like

```
if o.processors.NodeGroupPoolListProcessor != nil {
  nodeGroupPoolList = o.processors.NodeGroupPoolListProcessor.ProcessPool(//arguments)
} else if o.processors.NodeGroupPoolListProcessor != nil {
  nodeGroupPoolList = [][]NodeGroup{o.processors.NodeGroupPoolListProcessor.Process(//arguments)}
} else {
  nodeGroupPoolList = [][]NodeGroup{nodeGroups}
}
```

What do you think? It seems a bit less ambiguous to me, we can be always certain what path is the implementation taking after it is initialised.

Reply:
Makes sense. Done.